src/clp_ffi_py/ir/native/PyKeyValuePairLogEvent.cpp

```diff
@@ -0,0 +1,339 @@
+        ":param dictionary: A dictionary representing the key-value log event, where all keys are"
+        " expected to be of string type, including keys inside any sub-dictionaries.\n"
+);
+CLP_FFI_PY_METHOD auto PyKeyValuePairLogEvent_init(
+        PyKeyValuePairLogEvent* self,
+        PyObject* args,
+        PyObject* keywords
+) -> int;
+
+/**
+ * Callback of `PyKeyValuePairLogEvent`'s `to_dict` method.
+ */
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays)
+PyDoc_STRVAR(
+        cPyKeyValuePairLogEventToDictDoc,
+        "to_dict(self)\n"
+        "--\n\n"
+        "Converts the underlying key-value pair log event into a Python dictionary.\n\n"
+        ":return: Serialized log event in a Python dictionary.\n"
+);
+CLP_FFI_PY_METHOD auto PyKeyValuePairLogEvent_to_dict(PyKeyValuePairLogEvent* self) -> PyObject*;
+
+/**
+ * Callback of `PyKeyValuePairLogEvent`'s deallocator.
+ */
+CLP_FFI_PY_METHOD auto PyKeyValuePairLogEvent_dealloc(PyKeyValuePairLogEvent* self) -> void;
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays)
+PyMethodDef PyKeyValuePairLogEvent_method_table[]{
+        {"to_dict",
+         py_c_function_cast(PyKeyValuePairLogEvent_to_dict),
+         METH_NOARGS,
+         static_cast<char const*>(cPyKeyValuePairLogEventToDictDoc)},
+
+        {nullptr}
+};
+
+// NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, cppcoreguidelines-pro-type-*-cast)
+PyType_Slot PyKeyValuePairLogEvent_slots[]{
+        {Py_tp_alloc, reinterpret_cast<void*>(PyType_GenericAlloc)},
+        {Py_tp_dealloc, reinterpret_cast<void*>(PyKeyValuePairLogEvent_dealloc)},
+        {Py_tp_new, reinterpret_cast<void*>(PyType_GenericNew)},
+        {Py_tp_init, reinterpret_cast<void*>(PyKeyValuePairLogEvent_init)},
+        {Py_tp_methods, static_cast<void*>(PyKeyValuePairLogEvent_method_table)},
+        {Py_tp_doc, const_cast<void*>(static_cast<void const*>(cPyKeyValuePairLogEventDoc))},
+        {0, nullptr}
+};
+// NOLINTEND(cppcoreguidelines-avoid-c-arrays, cppcoreguidelines-pro-type-*-cast)
+
+/**
+ * `PyKeyValuePairLogEvent`'s Python type specifications.
+ */
+PyType_Spec PyKeyValuePairLogEvent_type_spec{
+        "clp_ffi_py.ir.native.KeyValuePairLogEvent",
+        sizeof(PyKeyValuePairLogEvent),
+        0,
+        Py_TPFLAGS_DEFAULT,
+        static_cast<PyType_Slot*>(PyKeyValuePairLogEvent_slots)
+};
+
+/**
+ * Converts the given Python dictionary to `clp::ffi::KeyValuePairLogEvent`.
+ * NOTE: Currently, there is no direct utility for constructing a `clp::ffi::KeyValuePairLogEvent`
+ * from a dictionary-like structure. As a workaround, this function serializes the dictionary
+ * using the key-value pair IR format, then deserializes it to create a `KeyValuePairLogEvent`
+ * instance. This approach is inefficient and intended solely for testing purposes, as it allows
+ * instance creation without a full IR stream. Future implementations should replace this method
+ * with a more efficient conversion once a direct utility is available.
```

Comment:
```suggestion
 * instance creation without a full IR stream. TODO: Replace this method with a more efficient
 * conversion once a direct utility is available.
```

I think it makes more sense to put the TODO here (and get rid of the one inside the implementation) so we don't forget to also update the docstring.